### PR TITLE
[avm1] Better handling of preload/suppress functions flags

### PR DESCRIPTION
**Pull request by moulins**
- Handle the case where both preload aud suppress flags are set for the same variable;
- Remove `arguments` field in `Activation`; instead use a normal local definition;
- When `suppress_this` is set, inherit the `this` value from parent activation. (This isn't entirely correct, as FP's `this` is mutable and seems to be part of the scope chain, but this would require a larger refactoring)

The added test contains hand-edited bytecode, so there's no accompanying `.as` file.

### CODE REVIEW

**Comment by Toad06** [comment]
This will fix #6204.

I also noticed a few regressions on files that are already known to have issues:

[Powerpuff Girls: Attack of the Puppybots](https://www.y8.com/games/powerpuff_girls_attack_of_the_puppybots)
- When the level starts, the characters and enemies no longer appear.
- Original issue: #5277

[Zelda Lampshade](https://www.newgrounds.com/portal/view/253819/format/flash?emulate=flash)
- When entering the dungeon, multiple instances of the boss are now created, as well as the music. Collisions also no longer work.
- Original issue: #1989

Maybe that these cases are just a side effect and would require `this` to be mutable?

**Comment by moulins** [comment]
Hmm, it's interesting that both of the original issues are caused by #1513. I should try merging #5492 on top of this to see if it fixes these games.

File: core/src/avm1/function.rs [hunk 1]
```diff
@@ -143,6 +143,136 @@ impl<'gc> Avm1Function<'gc> {
...
+            }
+        }
+        result.push(')');
+        result
+    }
+
+    fn is_this_inherited(&self) -> bool {
+        self.flags
+            .intersects(FunctionFlags::PRELOAD_THIS | FunctionFlags::SUPPRESS_THIS)
+    }
```
Comment (Herschel) [inline core/src/avm1/function.rs:166]: This seems a little overkill to as a function to me, as opposed to just using it inline.

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**